fbt_lib/src/lib.rs

```diff
@@ -1,13 +1,22 @@
+use crate::types::{Failure, SingleTestResult, TestResult};
 use std::path::PathBuf;
+use std::time::{Duration, Instant};
```

Comment:
Avoid `use` statements. Put entire path. You can make the path slightly better, eg put a `pub use types::*` in `lib.rs`, so you can do `pub fn test_all() -> anyhow::Result<crate::TestResult> {`.